content/blog/2022-07-28-switching-to-dvc-from-jupyter-vscode.md

```diff
@@ -0,0 +1,154 @@
+metrics for how well it performs. This file will have the `Set test variables`,
+`Load model and test data`, `Get model predictions`,
+`Calculate model performance metrics`, and `Save model performance metrics`
+notebook cells.
+
+That's it! Now you have all of the steps that you executed in your Jupyter
+notebook in a couple of files that you can easily edit and track across all of
+your experiments. This is a great time to commit these changes to your Git repo
+with the following commands:
+
+```cli
+$ git add train.py evaluate.py
+$ git commit -m "converted notebook to DVC project"
+```
+
+## Look at metrics from experiments
+
+Now we can create a DVC pipeline that executes these scripts to get the metrics
+for your experiments. If you look in the project's `dvc.yaml`, you'll see the
+stages we execute on an experiment run.
+
+```yaml
+stages:
+  train:
+    cmd: python src/train.py ./data/ ./models/model.pkl
+    deps:
+      - ./data/train.pkl
+      - ./src/train.py
+    params:
+      - train.seed
+      - train.n_est
+      - train.min_split
+    outs:
+      - ./models/model.pkl
+  evaluate:
+    cmd:
+      python ./src/evaluate.py ./models/model.pkl ./data scores.json prc.json
+      roc.json
+    deps:
+      - ./data
+      - ./models/model.pkl
+      - ./src/evaluate.py
+    metrics:
+      - scores.json:
+          cache: false
+    plots:
+      - prc.json:
+          cache: false
+          x: recall
+          y: precision
+      - roc.json:
+          cache: false
+          x: fpr
+          y: tpr
+```
+
+This runs everything in the same order that the Jupyter notebook did with a
+trackable structure. Now when you run `dvc exp run` to conduct an experiment,
+you can check out your metrics with either the CLI command `dvc exp show` or
+with
+[the DVC VSCode extension](https://marketplace.visualstudio.com/items?itemName=Iterative.dvc).
+
+```dvctable
+─────────────────────────────────────────────────────────────────────────────────────────────────────────────>
+  neutral:**Experiment**                neutral:**Created**        metric:**avg_prec**   metric:**roc_auc**   param:**train.seed**   param:**train.n_est**   param:**train.min_split**   >
+ ────────────────────────────────────────────────────────────────────────────────────────────────────────────>
+  **workspace**                 **-**               **0.76681**   **0.38867**   **20210428**     **300**           **75**                >
+  **jupyter-to-dvc**            **Jul 18, 2022**    **0.76681**   **0.38867**   **20210428**     **300**           **75**                >
+  └── 4a070a7 [exp-b8925]   Jul 18, 2022    0.76681   0.38867   20210428     300           75                >
+ ────────────────────────────────────────────────────────────────────────────────────────────────────────────>
+```
+
+_with CLI tool_
+
+![metrics in DVC VSCode extension](/uploads/images/2022-07-28/dvc-exp-in-vscode.png)
+
+_with DVC VSCode extension_
+
```

Comment:
Would it make sense to include any other steps?
* How to tweak and run new experiments
* How to retrieve a previous experiment

Reply:
Resolving as out of scope.